src/main/webapp/app/assessment/assessment.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Exercise } from 'app/entities/exercise.model';
+import { Complaint, ComplaintType } from 'app/entities/complaint.model';
+import { Result } from 'app/entities/result.model';
+
+/**
+ * For team exercises, the team tutor is the assessor and handles both complaints and feedback requests himself
+ * For individual exercises, complaints are handled by a secondary reviewer and feedback requests by the assessor himself
+ * For exam test runs, the original assessor is allowed to respond to complaints.
+ */
+export const isAllowedToRespondToComplaintAction = (isAtLeastInstructor: boolean, isTestRun: boolean, isAssessor: boolean, complaint: Complaint, exercise?: Exercise): boolean => {
+    if (isAtLeastInstructor) {
+        return true;
+    }
+    if (exercise?.teamMode || isTestRun) {
+        return isAssessor;
+    }
+    if (complaint.result && complaint.result.assessor == undefined) {
+        return true;
+    }
+    return complaint!.complaintType === ComplaintType.COMPLAINT ? !isAssessor : isAssessor;
+};
+
+/**
+ * During assessment, modifying the feedback should be allowed.
+ * After the assessment, modification should be prevented if the deadline hasn't passed yet.
```

Comment:
Thanks for pointing this out. I adapted it and added a test for it.

Regarding your other issue, are you sure, you tested it on the latest version of this PR? Because the tests show no indication.
I'll take a look tomorrow. A fix will be included if it wasn't merged before. It should not be a blocker.

Reply:
I checked and i was using 16a2fb9 which should have been the latest version on github at the time, but i didn't rebuild the server as this doesnt change it, so it was still on 3.2.2, so if some other pr changes how the locks are handled i can see how things could actually work correctly, ill re-check that (and the fix) later

Reply:
I was able to track your issue down. For some instances, the assessor was removed for tutors and editors. That's what caused the issue. I fixed this on the server and now it should hide everything and throw the correct error.